python/python/lance/dataset.py

```diff
@@ -942,6 +945,13 @@ def _coerce_reader(
         return pa.dataset.Scanner.from_dataset(data_obj).to_reader()
     elif isinstance(data_obj, pa.dataset.Scanner):
         return data_obj.to_reader()
+    elif isinstance(data_obj, Iterator):
+        if schema is not None:
+            return pa.RecordBatchReader.from_batches(schema, data_obj)
+        else:
```

Comment:
Can we use the schema from the first batch?

Reply:
From what I recall, we can't access that batch anymore if we expose it in the iterator right? Let me know if I'm misunderstanding.

Reply:
Yeah we'd have to implement a peekable iterator. I don't think we need to do that for now. I think beginners will simply write tables, and advanced users won't have much issue providing the schema themselves.